### ISSUE TITLE
Hash collision error after upgrading to 1.11.0

### ISSUE THREAD

**Issue by leamingrad** [ocaml/dune#2520]
Since upgrading to `1.11.0`, using the `-w` flag with `dune build` fails for me after the first build with:

```
Error: Hash collision between set of ppx drivers:
- cache : cs_lint (in project: cs)
- fetch : cs_lint (in project: cs)
```
Running `dune build` on its own works correctly. Any help on this issue would be appreciated.

Note: I think this might be a regression of the fix to #1524, but I'm raising a separate issue rather than reopening that one. I'm happy to just reopen that if it is easier.

**Comment by rgrinberg** [comment]
Do you have a project handy that we can use to reproduce this?

**Comment by leamingrad** [comment]
Not to hand sorry, but I will see if I can get a minimum example

**Comment by leamingrad** [comment]
Had a quick look with no luck, so I'll dig into this more later. For reference:
* The example at https://github.com/ocaml/dune/issues/1524#issuecomment-438935767 does not appear to trigger it any more
* `cs_lint` is a ppx rewriter that lives in the same repository, rather than being an external package

**Comment by diml** [comment]
I tried a few things but couldn't reproduce the issue. A reproduction case would be very welcome here :)

**Comment by leamingrad** [comment]
I have a MWE to reproduce this. The key seems to be using a ppx in a linting stanza.

I can reproduce the problem with the attached [dune-error-example.zip](https://github.com/ocaml/dune/files/3493215/dune-error-example.zip).

The error can be hit by running `dune build -w` in the root and then saving `collision.ml`.

This contains the following structure:

```
.
├── collision.ml
├── dune
├── dune-project
└── lint_ppx
    └── dune
```

`collision.ml` is an empty file, and the other files have the following contents:

`dune`:

```dune
(library
  (name collision)
  (modules collision)
  (lint (pps lint_ppx))
)
```

`dune-project`:

```dune
(lang dune 1.11)
```

`lint_ppx/dune`:

```dune
(library
  (name lint_ppx)
  (kind ppx_rewriter)
)
```

**Comment by diml** [comment]
I had a look, it's because we keep some values between builds that should be kept.

**Comment by diml** [comment]
Fixed in #2546, thanks for the reproduction case!

**Comment by leamingrad** [comment]
Excellent - thanks for the help with this issue

**Comment by diml** [comment]
This is now fixed in master

**Comment by jberdine** [comment]
Is there a plan to release this in a bug-fix release before 2.0?

**Comment by jberdine** [comment]
Is there a plan to release this in a bug-fix release before 2.0?

(Sorry for the duplicate comment, I realized that I was using 1.11.1 not 1.11.2 and assumed from the release date that this fix was included, but it appears not to be.)

**Comment by rgrinberg** [comment]
We can make 1.11.3 with this fix.

**Comment by diml** [comment]
I backported the fix in the 1.11 branch
